opencensus-shim/src/main/java/io/opentelemetry/api/trace/ProxyingSpan.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.trace;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Delegates <i>all</i> {@link Span} methods to some underlying Span via {@link
+ * ProxyingSpan#getProxied()}.
+ *
+ * <p>If not all calls are proxied, some, such as and in particular {@link
+ * Span#storeInContext(Context)} and {@link Span#makeCurrent()}, will use {@code this} instead of
+ * the proxied {@link Span} which betrays the expectation of instance fidelity imposed by the
+ * underlying otel mechanisms which minted the original {@link Span}, such as the otel javaagent.
+ *
+ * <p>This proxy class simplification allows the shim to perform its duties as minimally invasively
+ * as possible and itself never expose its own classes and objects to callers or recipients of calls
+ * from the shim.
+ *
+ * <p>This addresses the inconsistency where not all methods are appropriately delegated by exposing
+ * a single method, {@link ProxyingSpan#getProxied()}, to simplify and better ensure delegation and
+ * meeting expectations.
+ */
+// todo make this unnecessary
+@SuppressWarnings("UngroupedOverloads")
+public interface ProxyingSpan extends Span {
```

Comment:
this might be clearer as an abstract class

Reply:
I agree, but in this case it can't be as written: the opencensus shim code extends `io.opencensus.trace.Span` which is itself an abstract class. I suppose there's other ways of working that out, but as written it's not as easy as just changing it out.

Want me to rework this into an abstract class all the same? It may involve changes upstream though just given how it gets used -- hard to say without digging in.

Reply:
would this work?

`abstract class ProxyingSpan extends io.opencensus.trace.Span implements io.opentelemetry.api.trace.Span`

also, does `ProxyingSpan` need to be public?

Reply:
Good point: in this case, no I don't think it needs to be public. (See original thinking below)

That should work, yes. I'll make the change.

My original thinking on this was that `ProxyingSpan` could be used by the opentracing setup as well -- necessitating keeping it as decoupled from anything in opencensus -- if only to ensure all shims were correctly proxying the calls to the various otel Span, etc. methods to their agent-minted `AgentSpan` objects. But, stepping back that seems 1) ahead of myself and 2) acting without evidence that opentracing has a similar issue (I suspect it _might_, but it can also be fixed later).

So at this point I'm happy with either approach. But do lmk if you concur with my original thinking and think this should be pulled out and made usable for any other shims.

Reply:
I spoke too soon: there's a conflict between the `end()` methods of the OC and otel `Span` types: OC's is marked final giving us no way to intercept that call for the otel side without plugging in the final proxying method, `end()` into the implementation... Bit of a betrayal of the intent I had in mind, I fear, and muddies the implementation details by blurring the responsibility.

Do you still think the abstract class approach is worth it? I'm not so sure anymore and would personally want to keep the types decoupled, but I'm a newcomer to this project and am happy to defer.